conan/tools/build/cpu.py

```diff
@@ -5,13 +5,10 @@
 
 
 def build_jobs(conanfile):
-    njobs = conanfile.conf["tools.build:jobs"]
-    if njobs is not None:
-        result = int(njobs)
-        if result == 0:
-            return None
-        return result
-    return _cpu_count()
+    njobs = conanfile.conf.get("tools.build:jobs",
+                               default=_cpu_count(),
+                               check_type=int)
+    return njobs or None
```

Comment:
This is confusing, why ``njobs or None``? Shouldn't be it at least cpu_count?
Or is it just in case someone define "0"?

Reply:
Yes, the second one, because it had this original code:

```python
     result = int(njobs)
     if result == 0:
          return None
```
if someone defines `0` then we return `None`.

Reply:
It seemed that all the places where it was being used were prepared to check if it was returning more than 0 jobs. Removed 👍